Add Logging, External Configuration File, Minor Readability Changes

yellowmix opened the pull request:
logging module is used to provide two levels of output, INFO and DEBUG,
the latter being verbose within the counting loop.

configparser is used to enable use of an external configuration file.
This makes it easier to configure more options, such as output filename.

Set membership test on line 35 is made more explicit.

While 'flair' and 'x' variables are out of each others' scope, 'flair'
replaces local variable name 'x' to avoid confusion.

yellowmix committed:
Add Logging, External Configuration File, Minor Readability Changes

logging module is used to provide two levels of output, INFO and DEBUG,
the latter being verbose within the counting loop.

configparser is used to enable use of an external configuration file.
This makes it easier to configure more options, such as output filename.

Set membership test on line 35 is made more explicit.

While 'flair' and 'x' variables are out of each others' scope, 'flair'
replaces local variable name 'x' to avoid confusion.

ericwoolard commented:
Awesome! Thanks for helping out! Only thing I'd like to see (and would help any mods that aren't familiar with python or code in general) is just a simple print and sleep if an exception is thrown while trying to create the Reddit instance. Since it may be more likely that someone would run into an `invalid_grant` error from incorrect credentials somewhere than, say, hitting a short hiccup with Reddit and receiving a HTTP/Request exception, the endless loop that would occur of `invalid_grant` errors could be a bit scary to someone that's not sure what they're looking at haha. 

I thought about just having a separate except clause for `prawcore.exceptions.OAuthException` to trigger this so that we could handle that specific case a bit better, but people would have to also install prawcore, so meh lol. Something like: 

```
    except Exception as e:
        logging.error('{0}ERROR: {1}{2}'.format(err_color, e, color_end))
        print('{0}If you received an \'invalid_grant\' error, make sure the user and pass are correct \nand that '
              'they are the same as the account you registered the script with.{1}'.format(info_color, color_end))
        print('{0}>>> Retrying in 20 seconds, or ctrl+c to stop the script...{1}'.format(info_color, color_end))
        time.sleep(20)
```

The args being passed to the formatter could just be some ANSI codes for colored prints. Could have `err_color = '\x1b[0;31m'` for red, `info_color = '\x1b[0;33m'` for yellow, and `color_end = '\x1b[0m'` to set back to default so that the logging is separated a bit..Thoughts?

yellowmix committed:
Add more specific error handling to authentication

Login catches authentication errors and provides helpful information,
as well as information on how to stop the script.

yellowmix commented:
I added more specific error handling to address that. prawcore.exceptions is no big deal since it comes with PRAW if you install via pip or manually.

ericwoolard commented:
Oh, cool! For some reason I was under the impression that it was separate. This looks good to me! Thanks :)
